app/src/main/java/com/duckduckgo/app/referral/AppInstallationReferrerStateListener.kt

```diff
@@ -88,22 +88,28 @@ class PlayStoreAppReferrerStateListener @Inject constructor(
     override fun onInstallReferrerSetupFinished(responseCode: Int) {
         val referrerRetrievalDurationMs = System.currentTimeMillis() - initialisationStartTime
         Timber.i("Took ${referrerRetrievalDurationMs}ms to get initial referral data callback")
+        try {
+            when (responseCode) {
+                OK -> {
+                    Timber.d("Successfully connected to Referrer service")
+                    val response = referralClient.installReferrer
+                    val referrer = response.installReferrer
+                    val parsedResult = appInstallationReferrerParser.parse(referrer)
+                    referralResultReceived(parsedResult)
+                }
+                FEATURE_NOT_SUPPORTED -> referralResultFailed(FeatureNotSupported)
+                SERVICE_UNAVAILABLE -> referralResultFailed(ServiceUnavailable)
+                DEVELOPER_ERROR -> referralResultFailed(DeveloperError)
+                SERVICE_DISCONNECTED -> referralResultFailed(ServiceDisconnected)
+                else -> referralResultFailed(UnknownError)
 
-        when (responseCode) {
-            OK -> {
-                Timber.d("Successfully connected to Referrer service")
-                val response = referralClient.installReferrer
-                val referrer = response.installReferrer
-                val parsedResult = appInstallationReferrerParser.parse(referrer)
-                referralResultReceived(parsedResult)
             }
-            FEATURE_NOT_SUPPORTED -> referralResultFailed(FeatureNotSupported)
-            SERVICE_UNAVAILABLE -> referralResultFailed(ServiceUnavailable)
-            DEVELOPER_ERROR -> referralResultFailed(DeveloperError)
-            SERVICE_DISCONNECTED -> referralResultFailed(ServiceDisconnected)
-            else -> referralResultFailed(UnknownError)
+
+            referralClient.endConnection()
```

Comment:
Because we are dealing with such unstable API, I wonder if we should try to execute `endConnection()` inside a `finally` block + another `try-catch` in case it fails (old java typical code when dealing with Files), or do you think this method is the only possible source of exception within this block of code?.